lib/candy_check/app_store/verifier.rb

```diff
@@ -44,6 +46,16 @@ def verify_subscription(receipt_data, secret = nil)
         fetch_receipt_information(receipt_data, secret)
       end
 
+      # Calls a subscription verification for the given input
+      # @param receipt_data [String] the raw data to be verified
+      # @param secret [string] the optional shared secret
+      # @return [SubscriptionReceipt] if successful
+      # @return [VerificationFailure] otherwise
+      def verify_subscription_with_full_response(receipt_data, secret = nil)
+        @verifier = FullSubscriptionVerification
```

Comment:
it's weird to have `@verifier` as instance state. It introduces coupling in the order of invocation of the public methods here, which can cause race conditions if run in parallel.